trusted/pal/src/pal.rs

```diff
@@ -0,0 +1,174 @@
+//! This module expresses a simple abstraction layer between the trusted host and
+//! trusted application it runs. The main goal is to abstract away details of the
+//! trusted hosting and support mutiple host implementations. The host implementations
+//! differ in their capabilities with some imposing significant restrictions but
+//! offering better security and privacy guarantees. Hence the abstraction layer
+//! focuses on the most restricted host and assumes restricted kernel.
+//!
+//! Restricted kernel limits the execution model to processing signals (or rather
+//! messages) received from the untrusted launcher over a communication channel. Given
+//! this limitation the abstraction layer assumes that the trusted application performs
+//! processing in response to the signal from the untrusted launcher which can be
+//! either a message or a clock tick. In other words the abstraction layer defines
+//! a poll based execution model driven by a signal generating loop in untrusted
+//! launcher.
+//!
+//! [Application] trait must be implemented by a concrete trusted application to
+//! receive signals from the trusted host originating from untrusted launcher.
+//!
+//! [Host] trait must be implemented by a concrete trusted host to expose its capabilities
+//! to the trusted application.
+
+use alloc::boxed::Box;
+use alloc::vec::Vec;
+use anyhow::Result;
+
+type Message = Vec<u8>;
+
+/// Represents a set of claims for the trusted host and application, provides capability
+/// sign and verify application data.
+pub trait Attestation {
```

Comment:
I think it could have a method for getting `Evidence`, but the `Attestation` trait shouldn't itself represent an evidence, since `Attestation` will represent an API to the Restricted Kernel.

It also could be renamed to `Attester` to comply with the RATS [standard](https://datatracker.ietf.org/doc/html/rfc9334#name-attester).